### PR TITLE
🌿 Setup Fern TypeScript SDK + NPM Publishing

### PR DESCRIPTION
**Pull request by dsinghvi**
@jessie-young This PR sets up a couple things
 - root level `fern/` folder where your API definition lives
 - `fern generate` will dump TypeScript folders in `sdk/typescript`
 - I have added a couple files to `sdk/typescript` folder to make it easy for you to publish (package.json, build.js, tsconfig.json, etc.). You will have to maintain these files (the perk of the [Pro Plan](https://www.buildwithfern.com/pricing) is that we do all of this management for you in the cloud). 
 - GitHub action that runs `fern check` on every push to make sure that your api definition is valid
 - GitHub action that publishes your TypeScript SDK whenever you tag a release

### CODE REVIEW

File: .github/workflows/fern.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Install Fern
+        run:  cd sdk && npm install -g fern-api
+
+      - name: Check Fern API is valid
+        run: fern check
+  
+  publish-sdks:
```
Comment (dsinghvi) [inline .github/workflows/fern.yml:18]: You need to setup a repository secret called `NPM_TOKEN` and make it available to this Github action.

File: sdk/typescript/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+node_modules/
+src/
+dist/
```
Comment (dsinghvi) [inline sdk/typescript/.gitignore:3]: Going back on what I said on the call -- I think it'll be easier to generate the code in ci and not have it committed to the repo

File: sdk/typescript/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+    "name": "@fern-api/imdb",
```
Comment (dsinghvi) [inline sdk/typescript/package.json:2]: You should update this to your coordinate

File: sdk/typescript/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "version": "0.0.9",
```
Comment (dsinghvi) [inline sdk/typescript/package.json:3]: Everytime you want to bump the version for your sdk you will have to manually bump this version (or you can setup some automation)

File: .github/workflows/fern.yml (hunk 1 above)
Comment (dsinghvi) [reply .github/workflows/fern.yml:18]: You will also need a `FERN_TOKEN` to run this code generation in CI. Dmed you one.

**Review by jessie-young — APPROVED** [review]

**Pull request merged**